### PR TITLE
Add DockerHub Build + make container builds & push conditional on successful tests/

### PR DESCRIPTION
**Pull request by KoalaGeo**
# Overview

Taking experience with https://github.com/geopython/pygeoapi/pull/801 and updating GH Workflows for pycsw. 

Add DockerHub build & push
Update GHCR & DockerHub build to require `main.yml` workflow to be successful
Update documentation

For the DockerHub build you'll need to add Repo Secrets for 
```
username: ${{ secrets.DOCKER_USERNAME }}
password: ${{ secrets.DOCKER_PASSWORD }}
```
image

# Contributions and Licensing

(as per https://github.com/geopython/pycsw/blob/master/CONTRIBUTING.rst#contributions-and-licensing)

- [x] I'd like to contribute [feature X|bugfix Y|docs|something else] to pycsw. I confirm that my contributions to pycsw will be compatible with the pycsw license guidelines at the time of contribution.
- [x] I have already previously agreed to the pycsw Contributions and Licensing Guidelines

### CODE REVIEW

**Comment by kalxas** [comment]
Thank you, I will review

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# Triggers a Docker workflow on completion of the "build" workflow but
+# pushes to DockerHub
+# Runs GHC unit tests before DockerHub push.
```
Comment (kalxas) [inline .github/workflows/docker.yml:3]: forgot a "GHC" here :)

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+#
+name: Docker Build
+
+on:
+  workflow_run:
+    workflows: ["build ⚙️"]
+    types: [completed]
+
+jobs:
+  # Single job now to build Docker Image, run GHC unit tests, and push to DockerHub
```
Comment (kalxas) [inline .github/workflows/docker.yml:15]: same here

**Comment by kalxas** [comment]
Minor comments, looks good.
We will need to update the secrets on GitHub, correct?

**Comment by KoalaGeo** [comment]
Made those corrections, good spot!

& yes, the admin will need to add secrets for `${DOCKER_USERNAME}` & `${DOCKER_PASSWORD}`

**Review by kalxas — APPROVED** [review]

**Pull request merged**